x-pack/plugins/observability/public/plugin.ts

```diff
@@ -144,6 +145,12 @@ export class Plugin
       const { renderApp } = await import('./application');
       // Get start services
       const [coreStart, pluginsStart, { navigation }] = await coreSetup.getStartServices();
+      // Register alerts metadata
+      const { registerAlertsTableConfiguration } = await import(
+        './config/register_alerts_table_configuration'
+      );
+      const { alertsTableConfigurationRegistry } = pluginsStart.triggersActionsUi;
+      registerAlertsTableConfiguration(alertsTableConfigurationRegistry);
```

Comment:
@academo Based on my comment above, you will need to register AlertsTable configuration twice, one for Overview page and one for Alerts page, which means, that you need to parameterize the storage key.

`registerAlertsTableConfiguration(alertsTableConfigurationRegistry, 'xpack.observability.alert.tableState')` // for alerts page

`registerAlertsTableConfiguration(alertsTableConfigurationRegistry, 'xpack.observability.overview.alert.tableState')` // for overview page

Or the `registerAlertsTableConfiguration` could take as a 2nd argument the page you want to configure for example:

`registerAlertsTableConfiguration(alertsTableConfigurationRegistry, 'overview')` and `registerAlertsTableConfiguration(alertsTableConfigurationRegistry, 'alerts')` and registerAlertsTableConfiguration should handle the logic of setting the proper localStorage key according to the specified page (xpack.observability.overview.alert.tableState and xpack.observability.alert.tableState accordingly)

Reply:
@mgiota and I talked about it for now, o11y decide that they just want the default columns for alert registration and we will manage a different storage for the user in case